compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -92,18 +91,6 @@ impl<'a> Resolver<'a> {
     crate fn report_errors(&mut self, krate: &Crate) {
         self.report_with_use_injections(krate);
 
-        for &(span_use, span_def) in &self.macro_expanded_macro_export_errors {
```

Comment:
Why change the way this lint is processed?

Reply:
To simplify things.
I think we can just report it right away, without stashing anywhere, especially given that `fn resolve_ident_in_module_unadjusted_ext` now have access to a `NodeId` for linting.

Reply:
Other errors reported there are `privacy_errors` and `ambiguity_errors`.
I see why `privacy_errors` are stashed - because they are deduplicated, but they can be deduplicated on the fly too.
I don't see why `ambiguity_errors` need to be stashed.
Maybe there were reasons in the past.

Ok, I think I'll revert this change and try to move all the errors here to eager reporting in a separate PR.